src/checkInput.js

```diff
@@ -0,0 +1,146 @@
+import { sumOfGaussians } from './shapes/sumOfGaussians';
+import { sumOfLorentzians } from './shapes/sumOfLorentzians';
+
+export function checkInput(data, peaks, options) {
+  let {
+    shape = { kind: 'gaussian' },
+    optimization = {
+      kind: 'lm',
+    },
+  } = options;
+
+  if (typeof shape.kind !== 'string') {
+    throw new Error('kind should be a string');
+  }
+
+  let kind = shape.kind.toLowerCase().replace(/[^a-z]/g, '');
+
+  let paramsFunc;
+  let defaultParameters;
+  switch (kind) {
+    case 'gaussian':
+      paramsFunc = sumOfGaussians;
+      defaultParameters = {
+        x: {
+          init: (peak) => peak.x,
+          max: (peak) => peak.x + peak.width * 2,
+          min: (peak) => peak.x - peak.width * 2,
+          gradientDifference: (peak) => peak.width * 2e-3,
+        },
+        y: {
+          init: (peak) => peak.y,
+          max: () => 1.5,
+          min: () => 0,
+          gradientDifference: () => 1e-3,
+        },
+        width: {
+          init: (peak) => peak.width,
+          max: (peak) => peak.width * 2,
+          min: (peak) => peak.width * 0.5,
+          gradientDifference: (peak) => peak.width * 2e-3,
+        },
+      };
+      break;
+    case 'lorentzian':
+      paramsFunc = sumOfLorentzians;
+      defaultParameters = {
+        x: {
+          init: (peak) => peak.x,
+          max: (peak) => peak.x + peak.width * 2,
+          min: (peak) => peak.x - peak.width * 2,
+          gradientDifference: (peak) => peak.width * 2e-3,
+        },
+        y: {
+          init: (peak) => peak.y,
+          max: () => 1.5,
+          min: () => 0,
+          gradientDifference: () => 1e-3,
+        },
+        width: {
+          init: (peak) => peak.width,
+          max: (peak) => peak.width * 2,
+          min: (peak) => peak.width * 0.5,
+          gradientDifference: (peak) => peak.width * 2e-3,
+        },
+      };
+      break;
+    case 'pseudovoigt':
+      paramsFunc = sumOfGaussianLorentzians;
+      defaultParameters = {
+        x: {
+          init: (peak) => peak.x,
+          max: (peak) => peak.x + peak.width * 2,
+          min: (peak) => peak.x - peak.width * 2,
+          gradientDifference: (peak) => peak.width * 2e-3,
+        },
+        y: {
+          init: (peak) => peak.y,
+          max: () => 1.5,
+          min: () => 0,
+          gradientDifference: () => 1e-3,
+        },
+        width: {
+          init: (peak) => peak.width,
+          max: (peak) => peak.width * 2,
+          min: (peak) => peak.width * 0.5,
+          gradientDifference: (peak) => peak.width * 2e-3,
+        },
+        mu: {
+          init: (peak) => (peak.mu !== undefined ? peak.mu : 0.5),
+          min: () => 0,
+          max: () => 1,
+          gradientDifference: () => 0.01,
+        },
+      };
+      break;
+    default:
+      throw new Error('kind of shape is not supported');
+  }
+
+  let x = data.x;
```

Comment:
Could you still check if we can get the same results by giving the right gradient value without rescaling. Would be nice not to have to make this rescaling.

Reply:
I did try with maxY / 1e-3 which is equivalent to 1e-3 when data is rescaling dividing by maxY, and the result is not equal. I could search more.